examples/DynamicFilterNetwork/steering-filter.py

```diff
@@ -0,0 +1,286 @@
+        self.theta = np.array(range(8) * 4) / 8. * 2 * np.pi
+        self.filters = np.array([ThetaImages.get_filter(t) for t in self.theta])
+        self.cc = 0
+
+    def _setup_graph(self):
+        self.pred = self.trainer.get_predictor(['theta'], ['pred_filter'])
+
+    def _trigger_epoch(self):
+
+        def n(x):
+            x -= x.min()
+            x /= x.max()
+            return x
+
+        o = self.pred([self.theta])
+
+        gt_filters = np.concatenate([self.filters[i, :, :] for i in range(8)], axis=0)
+        pred_filters = np.concatenate([o[0][i, :, :, 0] for i in range(8)], axis=0)
+
+        canvas = np.concatenate([n(gt_filters), n(pred_filters)], axis=1)
+        l = canvas.shape[0] // 2
+        canvas = np.concatenate([canvas[:l], canvas[l:]], axis=1)
+        canvas = cv2.resize(canvas[..., None] * 255, (0, 0), fx=10, fy=10)
+
+        self.trainer.monitors.put_image('filter_export', canvas)
+        # # you might also want to write these images to disk (as in the casestudy from the docs)
+        # cv2.imwrite("export/out%04i.jpg" % self.cc, canvas)
+        # self.cc += 1
+
+
+class Model(ModelDesc):
+    def _get_inputs(self):
+        # TODO: allow arbitrary batch sizes
+        return [InputDesc(tf.float32, (BATCH, ), 'theta'),
+                InputDesc(tf.float32, (BATCH, SHAPE, SHAPE), 'image'),
+                InputDesc(tf.float32, (BATCH, SHAPE, SHAPE), 'gt_image'),
+                InputDesc(tf.float32, (BATCH, 9, 9), 'gt_filter')]
+
+    def _parameter_net(self, theta, kernel_shape=9):
+        """Estimate filters for convolution layers
+
+        Args:
+            theta: angle of filter
+            kernel_shape: size of each filter
+
+        Returns:
+            learned filter as [B, k, k]
+        """
+        with argscope(LeakyReLU, alpha=0.2), \
+                argscope(FullyConnected, nl=LeakyReLU):
+            net = FullyConnected('fc1', theta, 64)
+            net = FullyConnected('fc2', net, 128)
+
+        pred_filter = FullyConnected('conv3', net, kernel_shape ** 2, nl=tf.identity)
+        pred_filter = tf.reshape(pred_filter, [BATCH, kernel_shape, kernel_shape, 1], name="pred_filter")
+        logger.info('parameter net output: {}'.format(pred_filter.get_shape().as_list()))
+        return pred_filter
+
+    def _build_graph(self, input_vars):
+        kernel_size = 9
+        theta, image, gt_image, used_filter = input_vars
+
+        image = image
+        gt_image = gt_image
+
+        theta = tf.reshape(theta, [BATCH, 1, 1, 1]) - np.pi
+        image = tf.reshape(image, [BATCH, SHAPE, SHAPE, 1])
+        gt_image = tf.reshape(gt_image, [BATCH, SHAPE, SHAPE, 1])
+
+        pred_filter = self._parameter_net(theta)
+        pred_image = dynamic_conv_filter('dfn', image, pred_filter, 1, kernel_size)
+
+        with tf.name_scope('visualization'):
+            def n(x):
+                x = x - tf.reduce_min(x)
+                x = x / tf.reduce_max(x)
+                return x
```

Comment:
tf.summary.image will do some kind of normalization when input is float. Maybe this is not necessary?